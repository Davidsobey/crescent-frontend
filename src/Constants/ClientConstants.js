const ClientConstants = {
  CREATE_REQUEST: 'CREATE_COURSE_REQUEST',
  CREATE_SUCCESS: 'CREATE_CLIENT_SUCCESS',
  CREATE_FAILURE: 'CREATE_CLIENT_FAILURE',

  GETALL_REQUEST: 'GETALL_CLIENT_REQUEST',
  GETALL_SUCCESS: 'GETALL_CLIENT_SUCCESS',
  GETALL_FAILURE: 'GETALL_CLIENT_FAILURE',

  GETBYID_REQUEST: 'GETBYID_CLIENT_REQUEST',
  GETBYID_SUCCESS: 'GETBYID_CLIENT_SUCCESS',
  GETBYID_FAILURE: 'GETBYID_CLIENT_FAILURE',

  SUBSCRIBE_REQUEST: 'SUBSCRIBE_CLIENT_REQUEST',
  SUBSCRIBE_SUCCESS: 'SUBSCRIBE_CLIENT_SUCCESS',
  SUBSCRIBE_FAILURE: 'SUBSCRIBE_CLIENT_FAILURE',

  DELETE_CLIENT_REQUEST: 'DELETE_CLIENT_REQUEST',
  DELETE_CLIENT_SUCCESS: 'DELETE_CLIENT_SUCCESS',
  DELETE_CLIENT_FAILURE: 'DELETE_CLIENT_FAILURE',
};

export default ClientConstants;
