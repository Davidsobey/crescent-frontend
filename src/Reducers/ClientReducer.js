import ClientConstants from '../Constants/ClientConstants';
import UserConstants from '../Constants/UserConstants';

function ClientReducer(state = {}, action) {
  switch (action.type) {
    case ClientConstants.CREATE_REQUEST:
      return {
        client: action.client,
      };
    case ClientConstants.CREATE_SUCCESS:
      return {
        client: action.client,
      };
    case ClientConstants.CREATE_FAILURE:
      return {};
    case ClientConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case ClientConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        clients: action.clients,
      });
    case ClientConstants.GETALL_FAILURE:
      return {
        clients: {},
      };
    case ClientConstants.GETBYID_REQUEST:
      return {
        loading: true,
      };
    case ClientConstants.GETBYID_SUCCESS:
      return Object.assign({}, state, {
        client: action.client,
      });
    case ClientConstants.GETBYID_FAILURE:
      return {
        client: {},
      };
    case ClientConstants.SUBSCRIBE_REQUEST:
      return {
        subcription: action.subscription,
      };
    case ClientConstants.SUBSCRIBE_SUCCESS:
      return {
        subcription: action.subscription,
      };
    case ClientConstants.SUBSCRIBE_FAILURE:
      return {};
    case UserConstants.LOGOUT:
      return {};
    default:
      return state;
  }
}

export default ClientReducer;
