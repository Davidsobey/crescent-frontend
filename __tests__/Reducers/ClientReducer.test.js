// Reducer to be tested
import ClientReducer from '../../src/Reducers/ClientReducer';
import ClientConstants from '../../src/Constants/ClientConstants';

describe('ClientReducer', () => {
  test('CREATE_SUCCESS', () => {
    const action = { 
      type: ClientConstants.CREATE_SUCCESS, 
      client: {id:1, name:'aaa'},
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETALL_SUCCESS', () => {
    const action = { 
      type: ClientConstants.GETALL_SUCCESS, 
      clients: [],
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETBYID_SUCCESS', () => {
    const action = { 
      type: ClientConstants.GETBYID_SUCCESS, 
      client: {id:1, name:'aaa'},
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('SUBSCRIBE_SUCCESS', () => {
    const action = { 
      type: ClientConstants.SUBSCRIBE_SUCCESS, 
      subscription: 1,
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOAD_CLIENT_SUCCESS', () => {
    const action = { 
      type: ClientConstants.LOAD_CLIENT_SUCCESS, 
      client: {id:1, name:'aaa'},
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETUSERENROLMENTS_SUCCESS', () => {
    const action = { 
      type: ClientConstants.GETUSERENROLMENTS_SUCCESS, 
      userEnrolments: [],
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETSUBSCRIPTIONS_SUCCESS', () => {
    const action = { 
      type: ClientConstants.GETSUBSCRIPTIONS_SUCCESS, 
      subscriptions: [],
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });

  test('EDIT_SUCCESS', () => {
    const action = { 
      type: ClientConstants.EDIT_SUCCESS, 
    };

    expect(ClientReducer(undefined, action)).toMatchSnapshot();
  });
  
});
