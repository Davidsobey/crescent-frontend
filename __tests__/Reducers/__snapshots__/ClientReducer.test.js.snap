// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientReducer CREATE_SUCCESS 1`] = `
Object {
  "client": Object {
    "id": 1,
    "name": "aaa",
  },
  "clientRoleId": undefined,
  "creating": false,
  "newClientId": 1,
}
`;

exports[`ClientReducer EDIT_SUCCESS 1`] = `
Object {
  "client_editing": false,
}
`;

exports[`ClientReducer GETALL_SUCCESS 1`] = `
Object {
  "clients": Array [],
  "loading": false,
}
`;

exports[`ClientReducer GETBYID_SUCCESS 1`] = `
Object {
  "client": Object {
    "id": 1,
    "name": "aaa",
  },
}
`;

exports[`ClientReducer GETSUBSCRIPTIONS_SUCCESS 1`] = `
Object {
  "subscriptions": Array [],
  "subscriptions_loading": false,
}
`;

exports[`ClientReducer GETUSERENROLMENTS_SUCCESS 1`] = `
Object {
  "loading": false,
  "userEnrolments": Array [],
}
`;

exports[`ClientReducer LOAD_CLIENT_SUCCESS 1`] = `
Object {
  "client": Object {
    "id": 1,
    "name": "aaa",
  },
  "client_loading": false,
}
`;

exports[`ClientReducer SUBSCRIBE_SUCCESS 1`] = `
Object {
  "subcription": 1,
  "subscribing": false,
}
`;
